updated to new URL & adapted to new content format

richardpenner opened the pull request:
At a glance, this PR attempts to restore functionality to this project by using the new MLB.com probable pitchers URL, and adapting to the new HTML structure at that URL. Additionally, I've updated to a newer `Lab` module, as the old one had memory leaks. Lastly, some additional tests were added. 

Possible issue: I kept the test date the same, but interestingly I am seeing 15 games on that day, whereas the old test was returning 14.

Feedback welcome!

richardpenner committed:
updated to new URL & adapted to new content format

jaw187 commented:
Since I don't have CI setup, can you just post results of `npm test`

richardpenner commented:
Passed but I see you have linting setup – I’ll do a cleanup. Node >= 8 OK? If so I’ll add that to package.json too.

jaw187 commented:
ya, Node 8 is fine

On Fri, Mar 22, 2019 at 6:43 PM richardpenner <notifications@github.com>
wrote:

> Passed but I see you have linting setup – I’ll do a cleanup. Node >= 8 OK?
> If so I’ll add that to package.json too.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jaw187/mlbprobablepitchers/pull/4#issuecomment-475806252>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA2DNSWzjgMvZxy0hBZGa5Wly-G9EGxbks5vZVyogaJpZM4cEEJS>
> .
>

richardpenner committed:
fixed linter errors, added a few more tests, bumped minor version

richardpenner commented:
Tests passing, fixed all linter errors. Code coverage is set to 100, and I added a few tests, but the coverage is still not complete – any ideas?

<img width="996" alt="Screen Shot 2019-03-22 at 9 53 53 PM" src="https://user-images.githubusercontent.com/126490/54860186-0799f080-4ced-11e9-8a47-bdfcb48b52a4.png">

jaw187 commented on index.js:
```diff
@@ -1,79 +1,99 @@
 
-const internals = {};
+        let dateComponents = dateString.split('-');
+        if (dateComponents.length !== 3) {
```
Write a test to cover this code

jaw187 replied:
Nevermind, tests look good.  Not sure what's up with the coverage report.

richardpenner replied:
Yeah, I tried to add coverage, and it seems like I did? But still showing as not covered.

jaw187 commented on index.js:
```diff
@@ -1,79 +1,99 @@
-                    name: names[i],
-                    throws: throws[i]
+            const matches = pitcherIDInURL.exec(pitcher);
+            if ( matches.length !== 2 ) {
```
white space on the `(` and `)`

jaw187 commented on index.js:
```diff
@@ -1,79 +1,99 @@
-                    id: pitchers[j],
-                    name: names[j],
-                    throws: throws[j]
+                pitchers: {
```
This technically makes this a breaking change, but it makes sense

richardpenner replied:
sorry – which part breaks it? I'd intended to keep it backward compatible

richardpenner replied:
To be clear, I don't think the JSON format has changed at all – please let me know if it has.

jaw187 replied:
The object originally only had `home` and `away` properties that had the pitcher details.  Now those two objects are moved into a `pitchers` property.  I don't hate the change, it makes sense

jaw187 commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mlbprobablepitchers",
-  "version": "1.0.0",
+  "version": "1.1.0",
```
we'll version this after the code gets merged

jaw187 commented on test/index.js:
```diff
@@ -1,29 +1,71 @@
+        Probables.get(day, (err, dailyMatchups) => {
 
-describe('MLB Probable Pitchers', () => {
+            try {
```
Is this a new pattern for Lab v15?

jaw187 reviewed (changes requested):
Only required change is the white spaces.  But I'm worried that wrapping the test assertions in try/catch is not needed.

richardpenner committed:
fixed whitespace, removed try/catch around tests, removed version bump

jaw187 commented:
remove all of the try catches and copy/paste the test results, sorry

richardpenner committed:
removed try/catch around tests

richardpenner commented:
> mlbprobablepitchers@1.0.0 test /Users/richardpenner/PROJECTS/mlbprobablepitchers
> lab -m 9999 -v -L -t 100

MLB Probable Pitchers
  ✔ 1) Gets probables using date syntax 2011/07/23 (46 ms)
  ✔ 2) Gets probables using date syntax 2011-07-23 (2 ms)
  ✔ 3) Fails with invalid date syntax 2011-07 (1 ms)


3 tests complete
Test duration: 61 ms
No global variable leaks detected
Coverage: 68.67% (26/83)
index.js missing coverage on line(s): 8, 10, 11, 34, 35, 36, 37, 38, 44, 46, 47, 52, 54, 55, 57, 58, 59, 61, 63, 64, 65, 67, 68, 69, 92, 95
Code coverage below threshold: 68.67 < 100
Linting results: No issues

richardpenner commented:
Pushed an update removing the remaining try/catch blocks – I missed those earlier. Note that I added them because I was seeing some REALLY strange behaviour from Xray – when the test was failing, it invoked the Xray callback a second time. I'm not sure I was using the right node version, and I was also invoking the test directly rather than through `npm test`. Anyhow, I think it's working fine now, though the code coverage report is still inaccurate.
